include/aspect/simulator_access.h

```diff
@@ -937,20 +937,36 @@ namespace aspect
       n_particle_worlds() const;
 
       /**
-       * Returns a const reference to the particle world, in case anyone
+       * Returns a const reference to the vector containing all particle worlds,
+       * in case anyone wants to query something about particles.
+       */
+      const std::vector<std::unique_ptr<Particle::World<dim>>> &
+      get_particle_worlds() const;
```

Comment:
It doesnt look like you need these functions (`get_particle_worlds`) at the moment, they are not called. I would be happy if we could do without them, because they expose the internal data structure of how we store the particle worlds. And why would we need a non-const version? `SimulatorAccess` is supposed to give only read access.